python-bindings/overview_article/tasks.py

```diff
@@ -1,48 +1,95 @@
 """ Task definitions for invoke command line utility for python bindings
-    overview article. """
+    overview article.
+
+    If anything is causing trouble get in touch at
+    Znunu @ github or VimVim @ fcZBB2v @ discord"""
 import cffi
 import invoke
 import pathlib
+import sys
+import os
+import re
+
+on_win = sys.platform.startswith("win")
 
 
 @invoke.task
 def clean(c):
     """ Remove any built objects """
-    for pattern in ["*.o", "*.so", "cffi_example* cython_wrapper.cpp"]:
-        c.run("rm -rf {}".format(pattern))
+    for pattern in (
+        "*.o",
+        "*.so",
+        "*.obj",
+        "*.dll",
+        "*.exp",
+        "*.lib",
+        "*.pyx",
+        "*.pyd",
+        "cffi_example*",
+        "cython_wrapper.cpp",
+    ):
+        if on_win:
+            c.run("del {} >nul 2>&1".format(pattern))
+        else:
+            c.run("rm -rf {}".format(pattern))
+    if on_win:
+        c.run("rmdir /s /q Release >nul 2>&1")
 
 
 def print_banner(msg):
     print("==================================================")
     print("= {} ".format(msg))
 
 
-@invoke.task
-def build_cmult(c):
+@invoke.task()
+def build_cmult(c, path=None):
     """ Build the shared library for the sample C code """
-    print_banner("Building C Library")
-    invoke.run("gcc -c -Wall -Werror -fpic cmult.c -I /usr/include/python3.7")
-    invoke.run("gcc -shared -o libcmult.so cmult.o")
-    print("* Complete")
+    # Moving this type hint into signature causes an error (???)
+    c: invoke.Context
+    if on_win:
+        if not path:
+            print("Path is missing")
+        else:
+            # Using c.cd didn't work with paths that have spaces :/
+            path = f'"{path}vcvars32.bat" x86'  # Enter the VS venv
+            path += f'&& cd "{os.getcwd()}"'  # Change to current dir
+            path += "&& cl /LD cmult.c"  # Compile
+            # Uncomment line below, to suppress stdout
+            # path = path.replace("&&", " >nul &&") + " >nul"
+            c.run(path)
+    else:
+        print_banner("Building C Library")
+        cmd = "gcc -c -Wall -Werror -fpic cmult.c -I /usr/include/python3.7"
+        invoke.run(cmd)
+        invoke.run("gcc -shared -o cmult.so cmult.o")
+        print("* Complete")
 
 
-@invoke.task(build_cmult)
+@invoke.task()
```

Comment:
Is the build on windows taking significant time?  If not, I'd prefer to leave the dependency here.  It makes the individual targets a bit more fool-proof.

Reply:
No, it doesn't. On windows, the problem is that it'll only build if we supply the argument pointing to visual studio.

Reply:
ahhh.  I get it.  That's fine.  I missed that point.